packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -470,28 +516,61 @@ export class AtomicSearchBox {
     );
   }
 
-  private renderSuggestions() {
-    const showSuggestions = this.hasSuggestions && this.isExpanded;
+  private async updateSuggestedQuery(q: string) {
+    await Promise.allSettled(
+      this.suggestions.map((suggestion) =>
+        promiseTimeout(
+          suggestion.onSuggestedQueryChange
+            ? suggestion.onSuggestedQueryChange(q)
+            : Promise.resolve(),
+          this.suggestionTimeout
+        )
+      )
+    );
+    this.suggestedQuery = q;
+  }
 
+  private renderSuggestions() {
+    if (!this.hasSuggestions) {
+      return null;
+    }
+    const leftElements = this.getSuggestionElements(this.leftSuggestions);
+    const rightElements = this.getSuggestionElements(this.rightSuggestions);
     return (
-      <ul
-        id={this.popupId}
-        role="listbox"
-        part="suggestions"
-        aria-label={this.bindings.i18n.t('query-suggestion-list')}
-        ref={(el) => (this.listRef = el!)}
-        class={`w-full z-10 absolute left-0 top-full rounded-md bg-background border border-neutral ${
-          showSuggestions ? '' : 'hidden'
+      <div
+        class={`flex w-full z-10 absolute left-0 top-full rounded-md bg-background border border-neutral ${
+          this.showSuggestions ? '' : 'hidden'
         }`}
       >
-        {this.suggestionElements.map((suggestion, index) =>
-          this.renderSuggestion(
-            suggestion,
-            index,
-            this.suggestionElements.length - 1
-          )
+        {!!leftElements.length && (
+          <ul
+            id={this.popupId}
+            role="listbox"
```

Comment:
I'm not quite sure how to go about this aria stuff, never done it before! Could we have a chat tomorrow @btaillon ?

Reply:
Accessibility if very complex indeed !

For roles, the best resource I know is MDN: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles

And then you try to find one that describe best what the element does :D 

I think in this case it would be either... listbox or grid ?

What makes me hesitate for listbox is that it's not "static".

Reply:
I think a listbox is fine, so long as we have a single column of results and they aren't displayed as a grid.

My concern right now is the relationship between the two panes. If `rightElements` is strictly always genererated by the focused member of `leftElements`, then perhaps we should establish an [`aria-control`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls) relationship between left pane and the right pane. I have no idea